agent.go

```diff
@@ -327,14 +327,16 @@ func (a *Agent) startOnConnectionStateChangeRoutine() {
 }
 
 func (a *Agent) startConnectivityChecks(isControlling bool, remoteUfrag, remotePwd string) error {
-	if a.haveStarted.Load().(bool) {
+	a.muHaveStarted.Lock()
+	defer a.muHaveStarted.Unlock()
+	if a.haveStarted {
```

Comment:
Oops messed this up! Would you mind making this a swap instead? Maybe I am over enthuiastic, but I am really tired of debugging mutex/race issues :)

Then I would put the `SetRemoteCredentials` first. If you disagree happy to push forward :)

Reply:
Is it ok to `SetRemoteCredentials` before checking multiple call? I think it should keep old remote credentials which is already running.

Reply:
That's a good point, approved!